pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/invertedindex/TextIndexHandler.java

```diff
@@ -146,7 +147,10 @@ private boolean shouldCreateTextIndex(ColumnMetadata columnMetadata) {
     if (columnMetadata != null) {
       // Fail fast upon unsupported operations.
       checkUnsupportedOperationsForTextIndex(columnMetadata);
-      return true;
+
+      // Create text index only for those segments that are created within recent time-window
+      return _segmentMetadata == null
+          || _segmentMetadata.getIndexCreationTime() >= Instant.now().minusSeconds(3600).toEpochMilli();
```

Comment:
The method name is `getIndexCreationTime()`, but internally it returns `_creationTime`, which I think is the time when segment got created.

Reply:
Lets do as below:
lets add config `buildIndexForOldSegments` as config. 
```
"fieldConfigList":[
  {
     "name":"text_col_1",
     "encodingType":"RAW",
     "indexTypes": ["TEXT"],
     "properties":{"fstType":"native"},
     "buildForOldSegements" : false
  }
  ``` 
  
We can modify the `IndexLoadingConfig.java`.
Here we have one method :
```
privatee boolean extractBuildForOldSegementsForTextIndexFromFieldConfig(TableConfig tableConfig) {

}
```
```
private boolean shouldCreateTextIndex () {
  checkUnsupportedOperationsForTextIndex(columnMetadata) && indexLoadingConfig.buildForOldSegements()
}
```